code/game/gamemodes/cult/talisman.dm

```diff
@@ -1,3 +1,585 @@
+/datum/action/innate/cult/blood_magic //Blood magic handles the creation of blood spells (formerly talismans)
+	name = "Prepare Blood Magic"
+	button_icon_state = "carve"
+	desc = "Prepare blood magic by carving runes into your flesh. This rite is most effective with an <b>empowering rune</b>"
+	var/list/spells = list()
+	var/channeling = FALSE
+
+/datum/action/innate/cult/blood_magic/Grant()
+	..()
+	button.screen_loc = "6:-29,4:-2"
+	button.moved = "6:-29,4:-2"
+
+/datum/action/innate/cult/blood_magic/Remove()
+	for(var/X in spells)
+		qdel(X)
+	..()
+
+/datum/action/innate/cult/blood_magic/IsAvailable()
+	var/datum/antagonist/cult/C = owner.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+	if(!C)
+		return FALSE
+	return ..()
+
+/datum/action/innate/cult/blood_magic/proc/Positioning()
+	for(var/datum/action/innate/cult/blood_spell/B in spells)
+		var/pos = -29+spells.Find(B)*31
+		B.button.screen_loc = "6:[pos],4:-2"
+		B.button.moved = B.button.screen_loc
+
+/datum/action/innate/cult/blood_magic/Activate()
+	var/rune = FALSE
+	var/limit = RUNELESS_MAX_BLOODCHARGE
+	for(var/obj/effect/rune/empower/R in range(1, owner))
+		rune = TRUE
+		break
+	if(rune)
+		limit = MAX_BLOODCHARGE
+	if(spells.len >= limit)
+		if(rune)
+			to_chat(owner, "<span class='cultitalic'>Your body has reached its limit, you cannot store more than [MAX_BLOODCHARGE] spells at once. <b>Pick a spell to nullify.</b></span>")
+		else
+			to_chat(owner, "<span class='cultitalic'>Your body has reached its limit, <b>you cannot store more than [RUNELESS_MAX_BLOODCHARGE] spells at once without an empowering rune! Pick a spell to nullify.</b></span>")
+		var/nullify_spell = input(owner, "Choose a spell to remove.", "Current Spells") as null|anything in spells
+		if(nullify_spell)
+			qdel(nullify_spell)
+		return
+	var/entered_spell_name
+	var/datum/action/innate/cult/blood_spell/BS
+	var/list/possible_spells = list()
+	for(var/I in subtypesof(/datum/action/innate/cult/blood_spell))
+		var/datum/action/innate/cult/blood_spell/J = I
+		var/cult_name = initial(J.name)
+		possible_spells[cult_name] = J
+	entered_spell_name = input(owner, "Pick a blood spell to prepare...", "Spell Choices") as null|anything in possible_spells
+	BS = possible_spells[entered_spell_name]
+	if(QDELETED(src) || owner.incapacitated() || !BS)
+		return
+	to_chat(owner,"<span class='warning'>You begin to carve unnatural symbols into your flesh!</span>")
```

Comment:
and no one around them can see?

Reply:
They can see the dagger and the splash of blood when it finishes.